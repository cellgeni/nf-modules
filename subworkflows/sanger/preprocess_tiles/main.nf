include { IMAGING_PREPROCESS } from "../../../modules/sanger/imaging/preprocess"
include { IMAGING_GENERATECOMPANION } from "../../../modules/sanger/imaging/generatecompanion"

workflow PREPROCESS_TILES {
    take:
    experiments_ch // channel: [ val(meta), [ imaging_experiment ] ]
    psf_folder

    main:
    ch_versions = Channel.empty()

    IMAGING_GENERATECOMPANION(experiments_ch)
    ch_versions = ch_versions.mix(IMAGING_GENERATECOMPANION.out.versions.first())

    tiles = IMAGING_GENERATECOMPANION.out.csv
        .splitCsv(header: true, strip: true, sep: ",")
        .map { row ->
            [row[0], file(row[1].root_xml), row[1].image_id, row[1].index]
        }
    IMAGING_PREPROCESS(tiles, psf_folder)
    ch_versions = ch_versions.mix(IMAGING_PREPROCESS.out.versions.first())

    emit:
    companion_tiles = IMAGING_GENERATECOMPANION.out.companion.combine(IMAGING_PREPROCESS.out.fovs, by: 0) // [ meta, companion, [fovs] ]
    versions = ch_versions // channel: [ versions.yml ]
}
