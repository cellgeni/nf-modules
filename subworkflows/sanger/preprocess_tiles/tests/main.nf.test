// nf-core subworkflows test preprocess_tiles
nextflow_workflow {

    name "Test Subworkflow PREPROCESS_TILES"
    script "../main.nf"
    workflow "PREPROCESS_TILES"
    config "./nextflow.config"

    tag "subworkflows"
    tag "subworkflows_sanger"
    tag "subworkflows/preprocess_tiles"


    test("preprocess_tiles - xml - no_decon") {

        when {
            params {
                generatecompanion_args = "--master_file Index.idx.xml "
                preprocess_args = "--master_file Index.idx.xml"
            }
            workflow {
                """
                input[0] = [
                    [ id:'xml_no-decon' ], // meta map
                    0,
                    file(params.modules_testdata_base_path + 'PerkinElmer/Hoechst__2023-03-07T10_17_54-Measurement 2/Images/', checkIfExists: true),
                ]
                input[1] = file(params.modules_testdata_base_path + 'PSFs/', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }

    test("preprocess_tiles - xml - decon_10") {

        when {
            params {
                generatecompanion_args = "--master_file Index.idx.xml "
                preprocess_args = "--master_file Index.idx.xml -iterations 10"
            }
            workflow {
                """
                input[0] = [
                    [ id:'xml_decon' ], // meta map
                    0,
                    file(params.modules_testdata_base_path + 'PerkinElmer/Hoechst__2023-03-07T10_17_54-Measurement 2/Images/', checkIfExists: true),
                ]
                input[1] = file(params.modules_testdata_base_path + 'PSFs/', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }
}
