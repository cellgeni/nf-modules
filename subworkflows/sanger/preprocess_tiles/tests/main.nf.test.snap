{
    "preprocess_tiles - xml - decon_10": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "xml_decon"
                        },
                        "xml_decon_image0.companion.ome:md5,a7bd0dd0ba5df40099e5541dce2d793f",
                        "xml_decon_image0_Image:0.tif:md5,d28c2fbb2ec387ec2a9968ef1eaef093"
                    ]
                ],
                "1": [
                    "versions.yml:md5,6b57ad7e41b899248d7e0bbfe7c44cce",
                    "versions.yml:md5,9077b8085d6ac618f00810a072c96d26"
                ],
                "companion_tiles": [
                    [
                        {
                            "id": "xml_decon"
                        },
                        "xml_decon_image0.companion.ome:md5,a7bd0dd0ba5df40099e5541dce2d793f",
                        "xml_decon_image0_Image:0.tif:md5,d28c2fbb2ec387ec2a9968ef1eaef093"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,6b57ad7e41b899248d7e0bbfe7c44cce",
                    "versions.yml:md5,9077b8085d6ac618f00810a072c96d26"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-06-10T12:42:43.864446791"
    },
    "preprocess_tiles - xml - no_decon": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "xml_no-decon"
                        },
                        "xml_no-decon_image0.companion.ome:md5,b1afd7f6e54118beeb3048a0c7dcc8a7",
                        "xml_no-decon_image0_Image:0.tif:md5,d28c2fbb2ec387ec2a9968ef1eaef093"
                    ]
                ],
                "1": [
                    "versions.yml:md5,6b57ad7e41b899248d7e0bbfe7c44cce",
                    "versions.yml:md5,9077b8085d6ac618f00810a072c96d26"
                ],
                "companion_tiles": [
                    [
                        {
                            "id": "xml_no-decon"
                        },
                        "xml_no-decon_image0.companion.ome:md5,b1afd7f6e54118beeb3048a0c7dcc8a7",
                        "xml_no-decon_image0_Image:0.tif:md5,d28c2fbb2ec387ec2a9968ef1eaef093"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,6b57ad7e41b899248d7e0bbfe7c44cce",
                    "versions.yml:md5,9077b8085d6ac618f00810a072c96d26"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-06-10T12:41:20.377828645"
    }
}